pip/commands/install.py

```diff
@@ -255,7 +241,8 @@ def run(self, options, args):
                     build_dir=build_dir,
                     src_dir=options.src_dir,
                     download_dir=options.download_dir,
-                    upgrade=options.upgrade,
+                    upgrade=self.upgrade_option and options.upgrade,
+                    upgrade_direct=self.upgrade_direct,
                     as_egg=options.as_egg,
```

Comment:
again, maybe bikeshedding, here, but I want the RequirementSet properties to be meaningful for the long term

what we're doing here is keeping the general word "upgrade" to refer to the old style "recursive upgrade", and then tacking on a new property "upgrade_direct" for the new non-recursive type.

maybe that allows for a smaller PR diff, but I care more about readability.

to me the most readable properties for the Set would be:

`upgrade`:  whether it's an upgrade of any kind
`upgrade_recursive`: is it recursive or not